### PR TITLE
Testing for federation with NodeBB, make community.followers_url optional

### PR DESCRIPTION
**Pull request by Nutomic**
NodeBB doesnt have the `followers` field for community. This field is also not mandatory in the Activitypub standard, so we need to drop the requirement in Lemmy as well.

See https://community.nodebb.org/category/30/activitypub

### CODE REVIEW

File: crates/apub/src/objects/community.rs [hunk 1]
```diff
@@ -187,11 +187,9 @@ impl Object for ApubCommunity {
     let context_ = context.reset_request_count();
     spawn_try_task(async move {
       group.outbox.dereference(&community_, &context_).await.ok();
-      group
-        .followers
-        .dereference(&community_, &context_)
-        .await
-        .ok();
+      if let Some(followers) = group.followers {
+        followers.dereference(&community_, &context_).await.ok();
```
Comment (SleeplessOne1917) [inline crates/apub/src/objects/community.rs:191]: I'm not familiar with the `dereference` function and I couldn't find docs for it. I figure it's related to the dereference operator `*`?

Comment (dullbananas) [reply crates/apub/src/objects/community.rs:191]: Probably from a trait in the activitypub_federation crate

File: crates/db_schema/src/source/community.rs [hunk 2]
```diff
@@ -49,8 +49,8 @@ pub struct Community {
   /// A URL for a banner.
   pub banner: Option<DbUrl>,
   #[cfg_attr(feature = "full", ts(skip))]
-  #[serde(skip, default = "placeholder_apub_url")]
-  pub followers_url: DbUrl,
+  #[serde(skip)]
```
Comment (dessalines) [inline crates/db_schema/src/source/community.rs:52]: No breaking change since its skipped anyway.

**Review by dessalines — APPROVED** [review]

File: crates/apub/src/objects/community.rs (hunk 1 above)
Comment (Nutomic) [reply crates/apub/src/objects/community.rs:191]: https://docs.rs/activitypub_federation/latest/activitypub_federation/index.html#fetching-data

**Review by dullbananas — APPROVED** [review]

**Review by SleeplessOne1917 — APPROVED** [review]

**Pull request merged**